docs/develop/host-specific-api-model.md

```diff
@@ -0,0 +1,211 @@
+
+For example, the following code snippet declares the local JavaScript [Excel.Range](/javascript/api/excel/excel.range) object, `selectedRange`, to reference a selected range in the Excel workbook, and then sets some properties on that object. The `selectedRange` object is a proxy object, so the properties that are set and the method that is invoked on that object will not be reflected in the Excel document until your add-in calls `context.sync()`.
+
+```js
+var selectedRange = context.workbook.getSelectedRange();
+selectedRange.format.fill.color = "#4472C4";
+selectedRange.format.font.color = "white";
+selectedRange.format.autofitColumns();
+```
+
+### sync()
+
+Calling the `sync()` method on the request context synchronizes the state between proxy objects and objects in the Office document. The `sync()` method runs any commands that are queued on the request context and retrieves values for any properties that should be loaded on the proxy objects. The `sync()` method executes asynchronously and returns a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise), which is resolved when the `sync()` method completes.
+
+The following example shows a batch function that defines a local JavaScript proxy object (`selectedRange`), loads a property of that object, and then uses the JavaScript Promises pattern to call `context.sync()` to synchronize the state between proxy objects and objects in the Excel document.
+
+```js
+Excel.run(function (context) {
+    var selectedRange = context.workbook.getSelectedRange();
+    selectedRange.load('address');
+    return context.sync()
+      .then(function () {
+        console.log('The selected range is: ' + selectedRange.address);
+    });
+}).catch(function (error) {
+    console.log('error: ' + error);
+    if (error instanceof OfficeExtension.Error) {
+        console.log('Debug info: ' + JSON.stringify(error.debugInfo));
+    }
+});
+```
+
+In the previous example, `selectedRange` is set and its `address` property is loaded when `context.sync()` is called.
+
+Because `sync()` is an asynchronous operation that returns a promise, you should always `return` the promise (in JavaScript). Doing so ensures that the `sync()` operation completes before the script continues to run. For more information about optimizing performance with `sync()`, see [Excel JavaScript API performance optimization](../excel/performance.md).
```

Comment:
Maybe 
Because sync is an asynchronous operation, always return the promise to ensure the sync operation completes before the script runs

Reply:
Good call. I also added a bit about using `await` in TS/ES6+ JS.